src/Http/GraphRequest.php

```diff
@@ -50,7 +50,7 @@ class GraphRequest
     *
     * @var string
     */
-    protected $baseUrl;
+    protected $nationalCloud;
```

Comment:
What is the base url is for a custom endpoint? 

> Enable a developer to provide custom hosts.
CustomHosts option should be set on client creation.
CustomHosts option should be made available in the Context so that it is available to the middleware during request processing.
These hostnames are different from the Graph service endpoints on the national clouds.

https://github.com/microsoftgraph/msgraph-sdk-design/blob/master/GraphClientFactory.md

Reply:
I made a similar point. https://github.com/microsoftgraph/msgraph-sdk-php-core/pull/2#discussion_r660191608

The only time I've seen custom endpoints is for internal testing. I've seen zero demand for that with the PHP library. .NET and JavaScript, certainly. Maybe Java.

Reply:
I've added support for custom hosts that meet the criteria for a valid base URL